### PR TITLE
Mongoize/demongoize data elements 

### PR DESCRIPTION
**Pull request by okeefm**
Mongoize/demongoize data elements such that they can be stored in Arrays. Also add `_type` to some other basetypes.

Pull requests into cqm-models require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/CYPRESS-234
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] If there were any JavaScript changes, this PR has updated the `dist` directory using `npm run dist` N/A
- [x] If applicable, the library version number in `package.json` and `cqm-models.gemspec` has been updated

**Bonnie Reviewer:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Cypress Reviewer:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: app/models/qdm/basetypes/data_element.rb [hunk 1]
```diff
@@ -34,9 +34,54 @@ def id
...
     # Include '_type' in any JSON output. This is necessary for deserialization.
     def to_json(options = nil)
       serializable_hash(methods: :_type).to_json(options)
     end
+
+    class << self
+      # Get the object as it was stored in the database, and instantiate
+      # this custom class from it.
+      #
+      # The array elements in demongoize are the same 5 elements used in mongoize, i.e.
```
Comment (holmesie) [inline app/models/qdm/basetypes/data_element.rb]: This comment is not exactly correct anymore.

File: app/models/qdm/basetypes/data_element.rb [hunk 2]
```diff
@@ -34,9 +34,54 @@ def id
...
+      # [ low, high ].
+      def demongoize(object)
+        return nil unless object
+        object = object.symbolize_keys
+        if object.is_a?(Hash)
+          de = QDM::DataElement.new
```
Comment (holmesie) [inline app/models/qdm/basetypes/data_element.rb]: `de` - Germany? I'd like to see this spelled out.
